yellowbrick/features/jointplot.py

```diff
@@ -14,308 +14,386 @@
+                 hist=True, alpha=0.65, joint_kws=None, hist_kws=None, **kwargs):
+        # Initialize the visualizer
+        super(JointPlot, self).__init__(ax=ax, **kwargs)
+        self._xhax, self._yhax = None, None
+
+        # Set and validate the columns
+        self.columns = columns
+        if self.columns is not None and not isinstance(self.columns, (int, str)):
+            self.columns = tuple(self.columns)
+            if len(self.columns) > 2:
+                raise YellowbrickValueError((
+                    "'{}' contains too many indices or is invalid for joint plot"
+                ).format(columns))
+
+        # Seet and validate the correlation
+        self.correlation = correlation
+        if self.correlation not in self.correlation_methods:
+            raise YellowbrickValueError(
+                "'{}' is an invalid correlation method, use one of {}".format(
+                    self.correlation, ", ".join(self.correlation_methods.keys())
+            ))
 
-        Parameters
-        ----------
+        # Set and validate the kind of plot
+        self.kind = kind
+        if self.kind not in {'scatter', 'hex', 'hexbin'}:
+            raise YellowbrickValueError((
+                "'{}' is invalid joint plot kind, use 'scatter' or 'hex'"
+            ).format(self.kind))
 
-        X : ndarray or DataFrame of shape n x 1
-            A matrix of n instances with 1 feature
+        # Set and validate the histogram if specified
+        self.hist = hist
+        if self.hist not in {True, 'density', 'frequency', None, False}:
+                raise YellowbrickValueError((
+                    "'{}' is an invalid argument for hist, use None, True, "
+                    "False, 'density', or 'frequency'"
+                ).format(hist))
+
+        # If hist is True, test the version availability
+        if self.hist in {True, 'density', 'frequency'}:
+            self._layout()
+
+        # Set the additional visual parameters
+        self.alpha = alpha
+        self.joint_kws = joint_kws
+        self.hist_kws = hist_kws
+
+    @property
+    def xhax(self):
+        if self._xhax is None:
+            raise AttributeError(
+                "this visualizer does not have a histogram for the X axis"
+            )
+        return self._xhax
 
-        y : ndarray or Series of length n
-            An array or series of the target value
+    @property
+    def yhax(self):
+        if self._yhax is None:
+            raise AttributeError(
+                "this visualizer does not have a histogram for the Y axis"
+            )
+        return self._yhax
 
-        kwargs: dict
-            keyword arguments passed to Scikit-Learn API.
+    def _layout(self):
         """
+        Creates the grid layout for the joint plot, adding new axes for the histograms
+        if necessary and modifying the aspect ratio. Does nothing if hist is falsey.
```

Comment:
I actually did mean falsey here, but perhaps I could have communicated it better. It does nothing not only if `hist is False` but also if `hist is None` or `hist == ""`, etc. I guess I am using some [JavaScript terminology](http://adripofjavascript.com/blog/drips/truthy-and-falsy-values-in-javascript.html), though a quick Google reveals the term is used in Python too. I'll update to `if hist is False or None` since those are really the only two values I expect to be passed in as the arguments.